imports/plugins/included/payments-stripe/client/checkout/containers/injectedCardForm.js

```diff
@@ -0,0 +1,231 @@
+    const resubmitMessage = "Resubmit payment";
+    ev.preventDefault();
+
+    if (this.state.cardNumberErrorMessage || this.state.expDateErrorMessage || this.state.CVVErrorMessage || this.state.postalErrorMessage) {
+      return;
+    }
+
+    this.setState({
+      submitMessage: "Submitting...",
+      submitting: true
+    });
+    if (this.props.stripe) {
+      this.props.stripe.createToken().then((payload) => {
+        if (payload.error) {
+          this.setState({
+            errorMessage: "",
+            submitMessage: resubmitMessage,
+            submitting: false
+          });
+          // Do not duplicate field validation errors
+          if (payload.error.type !== "validation_error") {
+            this.setState({ errorMessage: payload.error.message });
+          }
+          return;
+        }
+        Meteor.apply("stripe/payment/createCharges", ["authorize", payload.token, this.props.cartId], {
+          onResultReceived: (error, result) => {
+            if (error || (result && result.error)) {
+              this.setState({
+                errorMessage: result.error.message,
+                submitMessage: resubmitMessage,
+                submitting: false
+              });
+            } else {
+              Meteor.call("cart/submitPayment", result.paymentMethods);
+              Router.go("cart/completed", {}, {
+                _id: this.props.cartId
+              });
+            }
+          }
+        });
+        return;
+      }).catch(() => {
+        this.setState({
+          errorMessage: "Something went wrong. Please try again.",
+          submitMessage: resubmitMessage,
+          submitting: false
+        });
+        return;
+      });
+    }
+    return;
+  };
+
+  handleCardNumberChange = (change) => {
+    if (change && change.error) {
+      this.setState({ cardNumberErrorMessage: change.error.message });
+    } else {
+      this.setState({ cardNumberErrorMessage: "" });
+    }
+  }
+
+  handleExpDateChange = (change) => {
+    if (change && change.error) {
+      this.setState({ expDateErrorMessage: change.error.message });
+    } else {
+      this.setState({ expDateErrorMessage: "" });
+    }
+  }
+
+  handleCVVChange = (change) => {
+    if (change && change.error) {
+      this.setState({ CVVErrorMessage: change.error.message });
+    } else {
+      this.setState({ CVVErrorMessage: "" });
+    }
+  }
+
+  handlePostalChange = (change) => {
+    if (change && change.value !== undefined) {
+      this.setState({ postal: change.value });
+    }
+    if (change && change.error) {
+      this.setState({ postalErrorMessage: change.error.message });
+    } else {
+      this.setState({ postalErrorMessage: "" });
+    }
+  }
+
+  displayErrorMessage = () => {
+    if (this.state.errorMessage) {
+      return (<div className="alert alert-danger">{this.state.errorMessage}</div>);
+    }
+    return null;
+  }
+
+  render() {
+    const style = {
+      base: {
+        "fontSize": "14px",
```

Comment:
I don't think we want styles hardcoded in here. I should be able to change styling without modifying core code.

Reply:
This is unavoidable as Elements API only allow font styles to be passed as props. I have put other non-font-related css, like borders and paddings, into a separate less file.